engine/src/main/java/org/destinationsol/world/generators/SolarSystemGenerator.java

```diff
@@ -324,6 +330,7 @@ protected void initializeRandomMazeGenerators() {
                 Class<? extends FeatureGenerator> newFeatureGeneratorType = featureGeneratorTypes.get(index);
                 try {
                     FeatureGenerator newFeatureGenerator = newFeatureGeneratorType.newInstance();
+                    ((MazeGenerator) newFeatureGenerator).setMazeConfigManager(context.get(MazeConfigs.class));
```

Comment:
The need for this cast (and later similar stuff) seems like a code smell of some sort. Near midnight so my brain is too hazy to describe it better 😅

Reply:
I think this one has to do with the way `Context` works